src/sage/schemes/curves/plane_curve_arrangement.py

```diff
@@ -0,0 +1,1259 @@
+from sage.schemes.curves.zariski_vankampen import fundamental_group_arrangement
+from sage.structure.category_object import normalize_names
+from sage.structure.parent import Parent
+from sage.structure.element import Element
+from sage.structure.richcmp import richcmp
+from sage.structure.unique_representation import UniqueRepresentation
+
+
+class AffinePlaneCurveArrangementsElement(Element):
+    """
+    An ordered affine plane curve arrangement.
+    """
+
+    def __init__(self, parent, curves, check=True):
+        """
+        Construct an ordered affine plane curve arrangement.
+
+        INPUT:
+
+        - ``parent`` -- the parent :class:`AffinePlaneCurveArrangements`
+
+        - ``curves`` -- a tuple of curves
+
+        EXAMPLES::
+
+            sage: H.<x, y> = AffinePlaneCurveArrangements(QQ)
+            sage: elt = H(x, y); elt
+            Arrangement (x, y) in Affine Space of dimension 2 over Rational Field
+            sage: TestSuite(elt).run()
+        """
+        super().__init__(parent)
+        self._curves = curves
+        if check:
+            if not isinstance(curves, tuple):
+                raise ValueError("the curves must be given as a tuple")
+            if not all(isinstance(h, AffinePlaneCurve) for h in curves):
+                raise ValueError("not all elements are curves")
+            if not all(h.ambient_space() is self.parent().ambient_space()
+                       for h in curves):
+                raise ValueError("not all curves are in the same ambient space")
+        self._braid_monodromy_non_vertical = None
+        self._braid_monodromy_vertical = None
+        self._strands_nonvertical = None
+        self._strands_vertical = None
+        self._fundamental_group_nonsimpl_nonvertical = None
+        self._fundamental_group_nonsimpl_vertical = None
+        self._fundamental_group_simpl_nonvertical = None
+        self._fundamental_group_simpl_vertical = None
+        self._meridians_nonsimpl_nonvertical = None
+        self._meridians_nonsimpl_vertical = None
+        self._meridians_simpl_nonvertical = None
+        self._meridians_simpl_vertical = None
+        self._vertical_lines_in_braid_mon = None
+
+    def __getitem__(self, i):
+        """
+        Return the `i`-th curve.
+
+        INPUT:
+
+        - ``i`` -- integer
+
+        OUTPUT:
+
+        The `i`-th curve.
+
+        EXAMPLES::
+
+            sage: H.<x, y> = AffinePlaneCurveArrangements(QQ)
+            sage: h = H(y^2 - x, y^3 + 2 * x^2, x^4 + y^4 + 1); h
+            Arrangement (y^2 - x, y^3 + 2*x^2, x^4 + y^4 + 1)
+            in Affine Space of dimension 2 over Rational Field
+        """
+        return self._curves[i]
+
+    def __hash__(self):
+        r"""
+        TESTS::
+
+            sage: H.<x, y> = AffinePlaneCurveArrangements(QQ)
+            sage: h = H((x * y, x + y +1))
+            sage: len_dict = {h: len(h)}
+        """
+        return hash(self.curves())
+
+    def n_curves(self):
```

Comment:
```suggestion
    def ncurves(self):
```
This would bring it in line with `ngens()` (and some other places in Sage IIRC).